### DISCUSSION TITLE
Skip routeRules for 404 pages?

### DISCUSSION THREAD

**Discussion by toooni** [nuxt/nuxt#16280] [Questions]
I do have a routeRule which adds a cache-control header to everything under `_nuxt`. This seems to work fine, but this rule is also applied if the response is a 404. How can I avoid this? Our CDN (CloudFront) rule sets the max-age to 10s but this is being overwritten by the 404 response which can result in cached 404 responses (file not ready during deploy).

```
routeRules: {
    "/_nuxt/**": { headers: { "cache-control": "max-age=31536000" } },
  },
```

**Comment by toooni** [comment]
I've fixed this issue for me/now with the following code in my error.vue:
```
const event = process.server && useRequestEvent();
if (event) {
  appendHeader(event, "Cache-Control", "max-age=0");
}
```

**Comment by TheAlexLichter** [comment]
This is definitely not intended 🙈
